cmdstanpy/cmds.py

```diff
@@ -79,140 +81,226 @@ def compile_model(
+        seeds.  If unspecified, numpy.random.RandomState() is used to generate
+        the seed. When the same seed is used across all chains, the chain-id
+        is used to advance the RNG to avoid dependent samples.
+
+    :param chain_ids: The offset for the random number generator, either
+        an integer or a list of unique per-chain offsets.  If unspecified,
+        chain ids are numbered sequentially starting from 1.
+
+    :param inits: Specifies how the sampler initializes parameter values.
+        Initializiation is either uniform random on a range centered on 0,
+        exactly 0, or a dictionary or file of initial values for some or all
+        parameters in the model.  The default initialization behavoir will
+        initialize all parameter values on range [-2, 2].  If these values
+        are too far from the expected parameter values, explicit initialization
+        may improve adaptation. The following value types are allowed:
+
+        * Single number `n > 0` - initialization range is [-n, n].
+        * `0` - all parameters are initialized to 0.
+        * dictionary - pairs parameter name : initial value.
+        * string - pathname to a JSON or Rdump file of initial parameter values.
+        * list of strings - per-chain pathname to data file.
+
+    :param warmup_iters: Number of iterations during warmup for each chain.
+
+    :param sampling_iters: Number of draws from the posterior for each chain.
+
+    :param warmup_schedule: Triple specifying fraction of total warmup
+        iterations allocated to each adaptation phase.  The default schedule
+        is (.15, .75, .10) where:
+
+        * Phase I is "fast" adaptation to find the typical set
+        * Phase II is "slow" adaptation to find the metric
+        * Phase III is "fast" adaptation to find the step_size.
+
+        For further details, see `the Stan Reference Manual, section
+            HMC algorithm parameters.
+
+    :param save_warmup: When True, sampler saves warmup draws as part of
+        the Stan csv output file.
+
+    :param thin: Period between saved samples.
+
+    :param max_treedepth: Maximum depth of trees evaluated by NUTS sampler
+        per iteration.
+
+    :param metric: Specification of the mass matrix.  The value is either
+        a string or a list of strings.  The strings `diag` and `dense`
+        specify whether or not to estimate the full covariance matrix or
+        only the diagonal elements.
+
+        The metric is initialized to a unit matrix.  If the value of the
+        metric argument is a string other than `diag` or `dense` or a
+        list of strings, these are treated as pathnames to a data file
+        in JSON or Rdump format which contains entry `inv_metric` which
+        is either a vector or a full matrix, depending on whether the
+        diagonal or dense covariance matrix is to be estimated.
+
+        The length of the list of pathnames must match the number of chains.
```

Comment:
pathnames -> paths (?)

Reply:
Also, what happens if the path is duplicated? On Windows that will probably crash, unless each file is copied to unique location (tempfile)

Reply:
no checks on pathname uniqueness - can add.

Reply:
list of `inits` arg paths must be unique as well, no?  adding checks on both args.